File: Data Structures & Algorithms/binary-tree-diameter/submission-2.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 **/

public class Solution {
    public int DiameterOfBinaryTree(TreeNode root) {
        
        if(root == null) return 0; 
        int leftCall = maxHeight(root.left);
        int rightCall = maxHeight(root.right);
        int diameter = leftCall + rightCall; 
        
        int res = Math.Max(DiameterOfBinaryTree(root.left), DiameterOfBinaryTree(root.right));
        int maggi = Math.Max(diameter, res);

        return maggi;

    }

    public int maxHeight(TreeNode root)
    {
        if (root == null) return 0;
        int dia = (Math.Max(maxHeight(root.left), maxHeight(root.right))) +1;
        return dia;
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-2.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        res = 0 
        for i in range(len(s)):
            contains = {}
            contains[s[i]] = (contains.get(s[i],0)+1)
            for j in range(i,len(s)):
                maxfreq= 0
                contains[s[j]] = (contains.get(s[j],0)+1)
                maxfreq = max(maxfreq, contains[s[j]])
                if((j-i+1)-maxfreq) <= k:
                    res = max(res, j-i+1)        
        return res